compiler/rustc_const_eval/src/transform/check_consts/ops.rs

```diff
@@ -438,6 +439,7 @@ impl<'tcx> NonConstOp<'tcx> for LiveDrop {
             "destructors cannot be evaluated at compile-time"
```

Comment:
Considering that we'll end up allowing destructors of some types, maybe just change the main message to say "destructor of type `Foo` can't be..."

Reply:
done